config/config.go

```diff
@@ -408,6 +433,48 @@ func (c *Config) InitializeWarpQuorums() error {
 	return nil
 }
 
+func (p *PChainAPI) Validate() error {
+	if _, err := url.ParseRequestURI(p.BaseURL); err != nil {
+		return fmt.Errorf("invalid base URL: %w", err)
+	}
+	p.options = make([]rpc.Option, 0, len(p.QueryParams))
+	for key, value := range p.QueryParams {
+		p.options = append(p.options, rpc.WithQueryParam(key, value))
+	}
+
+	p.client = platformvm.NewClient(p.BaseURL)
+	return nil
+}
+
+func (i *InfoAPI) Validate() error {
+	if _, err := url.ParseRequestURI(i.BaseURL); err != nil {
+		return fmt.Errorf("invalid base URL: %w", err)
+	}
+	i.options = make([]rpc.Option, 0, len(i.QueryParams))
+	for key, value := range i.QueryParams {
+		i.options = append(i.options, rpc.WithQueryParam(key, value))
+	}
```

Comment:
Just a stylistic nit, but I've often been burned by 1) initializing a slice with a specified _size_ and 2) appending to that slice in a loop. This results in a slice with the first half of the elements empty.

This isn't the case here, since you're initializing the slice with 0 _size_ and a specified _capacity_, but I had to double check to confirm this wasn't an instance of that bug.

I'd instead advocate for either `make` + assigning via index or `append`ing to an empty slice.

Reply:
Actually, I think we can simply remove `i.options = make([]rpc.Option, 0, len(i.QueryParams))`, since the empty slice will be allocated already.

Reply:
What he did is safe as far a he uses the append func. The only difference if he removes the `make` will be the underlying array of the slice being reallocated X times depending of the input array size. But as this action is done only once in the program and I suspect the array size to be low it is finally pretty the same.
But as you said it's stylistic, I don't want to question your review, just wanted to demystify the point. 👍

Reply:
yup what @najeal said. I'd pref initializing the slice with its capacity since we know the number of elements at initialization, and can help optimize not needing x memory allocations.

Reply:
Apologies if my original comment was misleading. For non-performance critical initialization code, code readability should take priority over small performance optimizations. In any case, I see this section has been removed.